prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -572,7 +602,25 @@ class ActionHistory {
         filteredOutReports.values.forEach { it.actionId = action.actionId }
         insertReports(txn)
         DatabaseAccess.saveTestData(covidMetaDataRecords, txn)
-        generateReportLineages(action.actionId)
+
+        // TODO: Generation of lineages needs to be separated from database insert functionality, since there
+        //  are conditional lineage generation rules
+        if (generatingEmptyReport) {
+            // if we are generating an empty report for the 'send' step there will be one report in and one out.
+            //  make sure to track the lineage. for the 'batch 'step there will not be any lineage
+            if (reportsIn.size == 1 && reportsOut.size == 1)
+                reportLineages.add(
+                    ReportLineage(
+                        null,
+                        action.actionId,
+                        reportsIn.values.first().reportId,
+                        reportsOut.values.first().reportId,
+                        null
+                    )
+                )
```

Comment:
Can this be moved into `trackGeneratedEmptyReport`? It might not if the normal `generateReportLineagesUsingItemLineage` call will break rather than being a noop

Reply:
This can not, based on the convoluted way and order of operations in insertAll.  This whole section needs to be refactored to not include logic within the save function, but this is the best place for it as the code is now.